pkg/operator/configobservation/node/observe_latency_profile.go

```diff
@@ -0,0 +1,70 @@
+package node
+
+import (
+	"fmt"
+
+	"github.com/openshift/cluster-kube-controller-manager-operator/pkg/operator/configobservation"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	configV1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+var nodeMonitorGracePeriodPath = []string{"extendedArguments", "node-monitor-grace-period"}
+
+// ObserveNodeMonitorGracePeriod observes the value that should be set for node-monitor-grace-period
+// controller manager argument on the basis of provided worker latency profile from config node object.
+func ObserveNodeMonitorGracePeriod(genericListers configobserver.Listers, _ events.Recorder, existingConfig map[string]interface{}) (ret map[string]interface{}, errs []error) {
+	defer func() {
+		// Prune the observed config so that it only contains node-monitor-grace-period field.
+		ret = configobserver.Pruned(ret, nodeMonitorGracePeriodPath)
+	}()
+
+	nodeLister := genericListers.(configobservation.Listers).NodeLister
+	node, err := nodeLister.Get("cluster")
+	if err != nil && !apierrors.IsNotFound(err) {
+		// we got an error so without the node object we are not able to determine worker latency profile
+		return existingConfig, append(errs, err)
+	}
+
+	// read the observed value
+	var observedNodeMonitorGracePeriod string
+	switch {
+	case node.Spec.WorkerLatencyProfile == configV1.DefaultUpdateDefaultReaction:
+		observedNodeMonitorGracePeriod = configV1.DefaultNodeMonitorGracePeriod.String()
+	case node.Spec.WorkerLatencyProfile == configV1.MediumUpdateAverageReaction:
+		observedNodeMonitorGracePeriod = configV1.MediumNodeMonitorGracePeriod.String()
+	case node.Spec.WorkerLatencyProfile == configV1.LowUpdateSlowReaction:
+		observedNodeMonitorGracePeriod = configV1.LowNodeMonitorGracePeriod.String()
+	// in case of empty worker latency profile, do not update config
+	case node.Spec.WorkerLatencyProfile == "":
```

Comment:
Yes that is intended behaviour. We can expect that the default value of this argument is very unlikely to change in the future however, in case it does we'll have to update/bump our code as per upstream in here as per upstream k8s.